usr/src/cmd/pkgsrc-wrapper/pkgsrc-wrapper.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2023 MNX Cloud, Inc.
+ */
+
+/*
+ * Use a C program instead of a shell script to exec() something in
+ * the pkgsrc paths.
+ */
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <errno.h>
+
+/* Remap table for when we have commands with than one name (e.g. tcsh/csh). */
+static struct {
+	const char *bin_name;
+	const char *bin_needs;
+} remap_table[] = {
+	{ "csh", "tcsh" },
+	/* login(1) invokes shells as -$SHELL. Cover that here. */
+	{ "-csh", "tcsh" },
+	{ "-tcsh", "tcsh" },
+	{ "-zsh", "zsh" },
+	{ NULL, NULL}
+};
+
+static const char *
+remap_bin(const char *base_bin)
+{
+	uint_t index;
+
+	for (index = 0; remap_table[index].bin_name != NULL; index++) {
+		/* Can use strcmp() because remap_table entries are bounded. */
+		if (strcmp(remap_table[index].bin_name, base_bin) == 0)
+			return (remap_table[index].bin_needs);
+
+	}
+	return (base_bin);
+}
+
+
+/* List these pkgsrc paths in desired search order. */
```

Comment:
Should probably make a note that disregarding `$PATH` is intentional.

Reply:
Will fix.